### PR TITLE
Remove old transport packages from CoreCLR and CoreFX.

### LINKED ISSUE #126: Remove private CoreFx and CoreClr transport packages
As we are switching away from package dependencies for CoreClr and CoreFx we should discuss remaining consumers of the private transport packages:
- https://github.com/dotnet/arcade/blob/d2d8d9f5c14e353acc7dfcf69c5fa5a8c368af40/src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.dependency.targets#L217-L218

I quickly talked with @dagood about this. The SharedFramework.Sdk requires crossgen from the CoreClr package and some OOB packages from the CoreFx one. We should probably package crossgen up and add it to BAR. For the OOB ones we already have a package which references all the OOB packages: `Microsoft.Private.CoreFx.OOB` which could probably be used instead.

cc @dotnet/runtime-infrastructure

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Now that dotnet/arcade#4708 is merged, the last user of a number of our old transport packages is gone. So, we should be able to remove most of our old transport packages:

- Microsoft.NETCore.Jit
- Microsoft.NETCore.Runtime.CoreCLR
- Microsoft.TargetingPack.Private.CoreCLR
- Microsoft.Private.CoreFx.NETCoreApp

We need to keep the IL packages since they're used via the IL SDK. We need to keep the Microsoft.NETCore.TestHost package since the diagnostic-tests repository still consumes it.

This PR also removes some now-dead code from the coreclr packaging infrastructure.

Fixes #126

### CODE REVIEW

**Review by trylek — APPROVED** [review]
Looks great to me, thank you!

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -1,9 +1,5 @@
 <Dependencies>
   <ProductDependencies>
-    <Dependency Name="Microsoft.Private.CoreFx.NETCoreApp" Version="5.0.0-alpha.1.19563.6">
```
Comment (jkotas) [inline eng/Version.Details.xml]: There are other places that have this references, e.g.: https://github.com/dotnet/runtime/blob/master/src/libraries/pkg/Microsoft.Windows.Compatibility/tests/Directory.Build.props#L9

Delete them too?

Comment (jkoritzinsky) [reply eng/Version.Details.xml]: I think that one might be required, but I'm not sure.

@ericstj any ideas if we still need the reference to Microsoft.Private.CoreFx.NETCoreApp in the Windows Compatibility Pack?

Comment (safern) [reply eng/Version.Details.xml]: The Microsoft.Windows.Compatibility pack tests are no longer used, we can delete those. They were used for the old infrastructure of producing APICatalog/Port ref assembly sets.

Comment (jkoritzinsky) [reply eng/Version.Details.xml]: I'll go remove them then!

**Review by hoyosjs — APPROVED** [review]
LGTM, although I just have some questions:
- What's going to happen with `libtracepointproviders.so` on unsupported architectures? Before we really excluded them from packaging. Tizen had issues with it for example.
- Now that the long-name generation of the DAC happens in CMake, will this still exist in the runtime packages?
- What's the packaging for the cross-arch DAC (and it's long name variant)? Right now they live under `artifacts\bin\coreclr\Windows_NT.arm64.Debug\x64\sharedFramework`

**Comment by jkoritzinsky** [comment]
> * What's going to happen with `libtracepointproviders.so` on unsupported architectures? Before we really excluded them from packaging. Tizen had issues with it for example.
> * Now that the long-name generation of the DAC happens in CMake, will this still exist in the runtime packages?
> * What's the packaging for the cross-arch DAC (and it's long name variant)? Right now they live under `artifacts\bin\coreclr\Windows_NT.arm64.Debug\x64\sharedFramework`

As is the case with all native CoreCLR files, if it gets installed in CMake to the sharedFramework folder, it will be packaged. If it is not installed to the sharedFramework folder, it won't be packaged.

This change makes no changes to the packages we ship.

**Review by ViktorHofer — APPROVED** [review]

File: src/libraries/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj [hunk 2]
```diff
@@ -1,78 +0,0 @@
...
-    <RefBinDir>$(NETCoreAppPackageRefPath)</RefBinDir>
-    <LibBinDir>$(NETCoreAppPackageRuntimePath)</LibBinDir>
-
-    <IsFrameworkPackage>true</IsFrameworkPackage>
-
-    <!-- Private packages need symbols -->
-    <IncludeSymbolsInPackage Condition="'$(IncludeSymbolsInPackage)' == ''">true</IncludeSymbolsInPackage>
-  </PropertyGroup>
-
-  <ItemGroup>
```
Comment (ericstj) [inline src/libraries/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]: @dagood do we have closure validation and type dup checking in final packages?

File: src/libraries/pkg/Microsoft.Private.CoreFx.NETCoreApp/netcoreapp.rids.props [hunk 3]
```diff
@@ -1,38 +0,0 @@
-﻿<Project>
-  <ItemGroup>
```
Comment (ericstj) [inline src/libraries/pkg/Microsoft.Private.CoreFx.NETCoreApp/netcoreapp.rids.props]: I bet you can delete framework package targets and other goo in pkg folder.

**Review by ericstj — COMMENTED** [review]
We should clean up https://github.com/dotnet/runtime/blob/151878260a9adf878a2d42ba084b6cd760dd51f9/src/libraries/Directory.Build.targets#L58.  Can we stop using the props file generation and instead just use simple file copies?

File: src/libraries/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj (hunk 2 above)
Comment (dagood) [reply src/libraries/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]: We should have it, but I'm not able to find evidence of it in an official build binlog... targets appear to noop and I don't see why at the moment. Filed https://github.com/dotnet/runtime/issues/31805.

**Comment by jkoritzinsky** [comment]
I don't know enough about the binplacing system in the libraries build to feel comfortable switching it to something different in this PR. I'd prefer to leave that to a follow up PR.

**Comment by ericstj** [comment]
> I don't know enough about the binplacing system in the libraries build to feel comfortable switching it to something different in this PR. I'd prefer to leave that to a follow up PR.

That’s fair.  We can follow up to change it.

**Comment by jkoritzinsky** [comment]
Test failure is #2176 
Merging.

**Pull request merged**